WcaOnRails/app/webpacker/components/CompetitionForm/ChampionshipInput.js

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState, useEffect, useContext } from 'react';
+import {
+  Button,
+  Form,
+  Icon,
+} from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import { championshipRegionsUrl } from '../../lib/requests/routes.js.erb';
+import Loading from '../Requests/Loading';
+import FormContext from './FormContext';
+
+function ChampionshipSelect({
+  inputState, index, regions, disabled,
+}) {
+  const localState = {
+    value: inputState.value[index].region,
+    onChange: (e, { value }) => {
+      const newValue = [...inputState.value];
+      newValue[index].region = value;
+      inputState.onChange(newValue);
+    },
+  };
+
+  const remove = () => {
+    const newValue = [...inputState.value];
+    newValue.splice(index, 1);
+    inputState.onChange(newValue);
+  };
+
+  return (
+    <Form.Field>
+      <label>{I18n.t('activerecord.attributes.championship.championship_type')}</label>
+      <Form.Group>
+        <Form.Select
+          width={6}
+          options={regions}
+          value={localState.value}
+          onChange={localState.onChange}
+          basic
+          disabled={disabled}
+        />
+        <Form.Button icon color="red" onClick={remove} disabled={disabled}>
+          <Icon inverted name="close" />
+        </Form.Button>
+      </Form.Group>
+    </Form.Field>
+  );
+}
+
+function getAvailibleId(data) {
+  const ids = data.map((championship) => championship.id);
```

Comment:
I would just do `Math.max(...data.map((championship) => championship.id));`
This is safe *just in case* there's a gap in the IDs and they are not all strictly consecutive.